homeassistant/components/switchbot/__init__.py

```diff
@@ -1 +1,80 @@
-"""The switchbot component."""
+"""Support for Switchbot devices."""
+import switchbot  # pylint: disable=import-error
+
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import (
+    CONF_RETRY_COUNT,
+    CONF_RETRY_TIMEOUT,
+    CONF_SCAN_TIMEOUT,
+    CONF_TIME_BETWEEN_UPDATE_COMMAND,
+    DATA_COORDINATOR,
+    DATA_UNDO_UPDATE_LISTENER,
+    DEFAULT_RETRY_COUNT,
+    DEFAULT_RETRY_TIMEOUT,
+    DEFAULT_SCAN_TIMEOUT,
+    DEFAULT_TIME_BETWEEN_UPDATE_COMMAND,
+    DOMAIN,
+)
+from .coordinator import SwitchbotDataUpdateCoordinator
+
+PLATFORMS = ["switch"]
+
+
+async def async_setup_entry(hass, entry) -> bool:
+    """Set up Switchbot from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    if not entry.options:
+        options = {
+            CONF_TIME_BETWEEN_UPDATE_COMMAND: DEFAULT_TIME_BETWEEN_UPDATE_COMMAND,
+            CONF_RETRY_COUNT: DEFAULT_RETRY_COUNT,
+            CONF_RETRY_TIMEOUT: DEFAULT_RETRY_TIMEOUT,
+            CONF_SCAN_TIMEOUT: DEFAULT_SCAN_TIMEOUT,
+        }
+
+        hass.config_entries.async_update_entry(entry, options=options)
+
+    switchbot.DEFAULT_RETRY_TIMEOUT = entry.options[CONF_RETRY_TIMEOUT]
+
+    # Store api in coordinator.
+    coordinator = SwitchbotDataUpdateCoordinator(
+        hass,
+        update_interval=entry.options[CONF_TIME_BETWEEN_UPDATE_COMMAND],
+        api=switchbot,
+        retry_count=entry.options[CONF_RETRY_COUNT],
+        scan_timeout=entry.options[CONF_SCAN_TIMEOUT],
+    )
```

Comment:
Why do you want to store the API in coordinator? Why not import it where needed?

Reply:
1) To allow the user to change api defaults in config flow (Retry timeout L38 - L39)
2) For easy access in all platforms.

I could import just the GetSwitchbotDevices class and then import on the api on each platform. The coordinator is shared between all platforms so I think this is the most efficient way to do this. (Python just stores a memory reference in each platform I think.)